pandas/core/frame.py

```diff
@@ -10935,7 +10950,23 @@ def _get_data() -> DataFrame:
             df = _get_data()
         if axis is None:
             return func(df.values)
-        elif axis == 1:
+
+        # if len(df._mgr) > 0:
+        #     common_dtype = find_common_type(list(df._mgr.get_dtypes()))
+        #     is_masked_ea = isinstance(common_dtype, BaseMaskedDtype)
+        #     is_np = isinstance(common_dtype, np.dtype)
+        # else:
+        #     common_dtype = None
+
+        # if axis == 1 and common_dtype and is_masked_ea:
+        #     data, mask = self._mgr.as_array_masked()
+        #     ea2d = common_dtype.construct_array_type()(data, mask)
+        #     result = ea2d._reduce(name, axis=axis, skipna=skipna, **kwds)
+        #     labels = self._get_agg_axis(axis)
+        #     result = self._constructor_sliced(result, index=labels, copy=False)
+        #     return result
```

Comment:
@rhshadrach this relates to https://github.com/pandas-dev/pandas/pull/51923, and making use of the `as_array_masked` added in this PR could be a potential different way of tackling it.

Manually running one of the benchmarks you mentioned in https://github.com/pandas-dev/pandas/pull/51955 (the bottom row of `-         8.94±0s       24.3±0.3ms     0.00  stat_ops.FrameOps.time_op('mean', 'Int64', 1)`), with the above uncommented, I get:

```
In [1]: values = np.random.randn(100000, 4)
   ...: df = pd.DataFrame(values.astype(int)).astype("Int64")

In [2]: %timeit df.mean(axis=1)
5.97 s ± 434 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)   # <-- main
1.74 ms ± 126 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)   # <-- PR
```

Now, this is probably a bit too optimistic, because the `as_array_masked` is currently assuming all the same dtypes, but with the use of `find_common_type` here, it would need to handle some extra cases as well. But in general it should also give a way to speed-up this specific case (while using the actual masked reduction implementations, which already support 2D with axis=1)

Reply:
Doing a cast beforehand might indeed be the easiest, then `as_array_masked` doesn't need to handle the case of multiple dtypes itself.